subprojects/gradle/android/src/main/kotlin/com/avito/android/AndroidSdk.kt

```diff
@@ -56,18 +58,16 @@ class AndroidSdk(
             return ExistingDirectory.Impl(dir)
         }
 
-    private val apkSigner = ApkSigner.Impl(buildToolsPath, processRunner)
+    private val apkSigner: ApkSigner
+        get() = ApkSigner.Impl(buildToolsPath, processRunner)
 
-    private val aapt = Aapt.Impl(buildToolsPath, processRunner)
+    private val aapt: Aapt
+        get() = Aapt.Impl(buildToolsPath, processRunner)
 
     private val keytool = KeyTool(processRunner)
 
     val androidHome: ExistingDirectory
-        get() = ExistingDirectory.Impl(File(System.getenv("ANDROID_HOME")
-            ?: androidHomeFromLocalProperties(project.rootProject.file("local.properties")) {
-                logger.error(it)
-            }
-            ?: error("Can't resolve ANDROID_HOME")))
+        get() = ExistingDirectory.Impl(project.androidHome)
```

Comment:
So I didn't get what did you do? What is the `project.androidHome`?

Reply:
It's the same function. It was a duplicate.

The difference is in another part. I've made the property lazy to avoid this logic in a configuration phase.

Reply:
I can't see a difference on laziness

Reply:
Laziness are only in `apkSigner` and `aapt` properties.
Sorry, I thought that the first question was more general.